web/writing.html

```diff
@@ -362,6 +367,144 @@ <h5 class="portlet__head">Webhook</h5>
+          'porkbun': document.getElementById('porkbun'),
+          'godaddy': document.getElementById('godaddy'),
+          'googledomain': document.getElementById('googledomain'),
+        },
+        'Ipv4GetType': {
+          'url': document.getElementById('urlRadioIpv4'),
+          'netInterface': document.getElementById('netInterfaceRadioIpv4'),
+          'cmd': document.getElementById('cmdRadioIpv4'),
+        },
+        'Ipv6GetType': {
+          'url': document.getElementById('urlRadioIpv6'),
+          'netInterface': document.getElementById('netInterfaceRadioIpv6'),
+          'cmd': document.getElementById('cmdRadioIpv6'),
+        },
+      };
+
+      document.getElementById('NotAllowWanAccess').checked = "{{.NotAllowWanAccess}}"=='on';
+      document.getElementById('Username').value = "{{.Username}}";
+      document.getElementById('Password').value = "{{.Password}}";
+      document.getElementById('WebhookURL').value = "{{.WebhookURL}}";
+      document.getElementById('WebhookRequestBody').value = "{{.WebhookRequestBody}}";
+      const df = new DocumentFragment();
+      for (const i in dnsconf) {
+        const opt = document.createElement('option');
+        opt.id = 'index_' + i;
+        opt.value = i;
+        opt.text = i;
+        df.appendChild(opt);
+      }
+      document.getElementById('index').appendChild(df);
+    });
+
+    window.addEventListener('load', () => {
+      document.getElementById('index').addEventListener('change', e => {
+        saveconf(index);
+        index = Number(e.target.value);
+        showconf(index);
+      });
+      document.getElementById('addbtn').addEventListener('click', e => {
+        e.preventDefault();
+        const confdefault = {
+          'DnsName': 'alidns',
+          'DnsID': '',
+          'DnsSecret': '',
+          'TTL': '',
+          'Ipv4Enable': 'on',
+          'Ipv4GetType': 'url',
+          'Ipv4Url': 'https://myip4.ipip.net,https://ddns.oray.com/checkip,https://ip.3322.net,https://4.ipw.cn',
+          'Ipv4Cmd': '',
+          'Ipv4Domains': '',
+          'Ipv6Enable': 'on',
+          'Ipv6GetType': 'netInterface',
+          'Ipv6Url': 'https://speed.neu6.edu.cn/getIP.php,https://v6.ident.me,https://6.ipw.cn',
+          'Ipv6Cmd': '',
+          'IPv6Reg': '',
+          'Ipv6Domains': '',
+        };
+        saveconf(index);
+        index = Number(dnsconf.length);
+        dnsconf[index] = confdefault;
+        const opt = document.createElement('option');
+        opt.id = 'index_' + index;
+        opt.value = index;
+        opt.text = index;
+        document.getElementById('index').appendChild(opt);
+        document.getElementById('index').value = index;
+        showconf(index);
+      });
+      document.getElementById('delbtn').addEventListener('click', e => {
+        e.preventDefault();
+        (document.getElementById('index_'+index)??{}).disabled = true;
+        dnsconf[index] = null;
```

Comment:
改成移除这个元素是不是更好些呢：

```suggestion
        dnsconf.splice(index, 1);
```

然后我觉得上文中弄成禁用这个选项并不好（应该从 dom 中移除这个元素），不然能看到又不能点，很奇怪。删除以后需要重新排一下索引（考虑加个函数专门来处理这些内容）

Reply:
根据之前作的讨论，前端暂不重排索引，以免增加匹配原有HideIDSecret的复杂度，保存时后端统一作处理。